node/test/PublicAPITest.ts

```diff
@@ -17,23 +17,135 @@ SignalClient.initLogger(
   }
 );
 
+class InMemorySessionStore extends SignalClient.SessionStore {
+  private state = new Map();
+  async saveSession(
+    name: SignalClient.ProtocolAddress,
+    record: SignalClient.SessionRecord
+  ): Promise<void> {
+    const idx = name.name() + '::' + name.deviceId();
+    Promise.resolve(this.state.set(idx, record.serialize()));
+  }
+  async getSession(
+    name: SignalClient.ProtocolAddress
+  ): Promise<SignalClient.SessionRecord | null> {
+    const idx = name.name() + '::' + name.deviceId();
+    if (this.state.has(idx)) {
+      return Promise.resolve(
+        SignalClient.SessionRecord.deserialize(this.state.get(idx))
+      );
+    } else {
+      return Promise.resolve(null);
+    }
+  }
+}
+
+class InMemoryIdentityKeyStore extends SignalClient.IdentityKeyStore {
+  private idKeys = new Map();
+  private localRegistrationId: number;
+  private identityKey: SignalClient.PrivateKey;
+
+  constructor() {
+    super();
+    this.identityKey = SignalClient.PrivateKey.generate();
+    this.localRegistrationId = 5;
+  }
+
+  async getIdentityKey(): Promise<SignalClient.PrivateKey> {
+    return Promise.resolve(this.identityKey);
+  }
+  async getLocalRegistrationId(): Promise<number> {
+    return Promise.resolve(this.localRegistrationId);
+  }
+
+  async isTrustedIdentity(
+    name: SignalClient.ProtocolAddress,
+    key: SignalClient.PublicKey,
+    _direction: SignalClient.Direction
+  ): Promise<boolean> {
+    const idx = name.name() + '::' + name.deviceId();
+    if (this.idKeys.has(idx)) {
+      const currentKey = this.idKeys.get(idx);
+      return Promise.resolve(currentKey.compare(key) == 0);
+    } else {
+      return Promise.resolve(true);
+    }
+  }
+
+  async saveIdentity(
+    name: SignalClient.ProtocolAddress,
+    record: SignalClient.PublicKey
+  ): Promise<boolean> {
+    const idx = name.name() + '::' + name.deviceId();
+    const seen = this.idKeys.has(idx);
+    if (!seen) {
+      this.idKeys.set(idx, record);
```

Comment:
This isn't quite correct: it's supposed to return `true` if the identity *changed,* but it's supposed to replace it regardless.